### PR TITLE
[ADD]account_loan_vendor_payment: open a loan directly from vendor pa…

### PR DESCRIPTION
**Pull request by odooNextev**
…yment

### CODE REVIEW

**Comment by francesco-ooops** [comment]
@etobella what do you think of this one?

File: account_loan_vendor_payment/models/account_loan.py
Comment (GSLabIt) [inline account_loan_vendor_payment/models/account_loan.py]: may be is better to reference `payment_ids` as loan is opened against the payment. What u think?

Comment (elvise) [reply account_loan_vendor_payment/models/account_loan.py]: @odooNextev gentle reminder :)

Comment (odooNextev) [reply account_loan_vendor_payment/models/account_loan.py]: I added payment_ids reference, but I didn't remove invoice_ids reference because is used to move from invoice to loan using a smart button.

Comment (GSLabIt) [reply account_loan_vendor_payment/models/account_loan.py]: > I added payment_ids reference, but I didn't remove invoice_ids reference because is used to move from invoice to loan using a smart button.

you can get invoice from payments, kind of `payments.mapped('invoice_id')`; would be enough?

Comment (odooNextev) [reply account_loan_vendor_payment/models/account_loan.py]: If I try to remove invoice_ids from model, I had some problem in account.move, because loan_ids is empty and I couldn't get loans from invoice, because payment_id reference is always empty.

Comment (GSLabIt) [reply account_loan_vendor_payment/models/account_loan.py]: i think you already have `move_ids` referencing `account.move` on `account.loan`, would that do the job without adding `invoice_ids`?

https://github.com/OCA/account-financial-tools/blob/fe88b783b1919c4e8cafbd436b4aefc13b3328ad/account_loan/model/account_loan.py#L225

File: account_loan_vendor_payment/models/account_payment.py
Comment (PicchiSeba) [inline account_loan_vendor_payment/models/account_payment.py:15]: **question**: why is this compute treated as if it's used by a recordset with a single item while the `_compute_get_loans_count` method it's implemented with more records in mind?

File: account_loan_vendor_payment/wizard/account_payment_register.py
Comment (PicchiSeba) [inline account_loan_vendor_payment/wizard/account_payment_register.py]: **nitpick**: maybe we should explicitly specify the default for both of them, for consistency

File: account_loan_vendor_payment/wizard/account_payment_register.py
Comment (PicchiSeba) [inline account_loan_vendor_payment/wizard/account_payment_register.py:25]: **question**: I'm probably missing a bit of context, could you explain why we're multiplying the amount by itself?

Comment (odooNextev) [reply account_loan_vendor_payment/wizard/account_payment_register.py:25]: Hi @PicchiSeba, exchange rate is an input value, so I multiply it to open a new loan using a different currency using the inserted exchange rate.

Comment (PicchiSeba) [reply account_loan_vendor_payment/wizard/account_payment_register.py:25]: I'm sorry i misinterpeted the expression.
At first I thought you wrote "multiply the amount by the exchange rate (if we have it), otherwise multiply it by itself"

But now I re-read the line and it's fine to me.

**Review by PicchiSeba — APPROVED** [review]
Code review: LGTM

File: account_loan_vendor_payment/models/account_payment.py
Comment (etobella) [inline account_loan_vendor_payment/models/account_payment.py:7]: Why are you making it stored?

You shouldn't do this, because all the other fields of the compute should have the same behaviour and it is obviously a bad decision IMO

File: account_loan_vendor_payment/models/account_payment.py
Comment (etobella) [inline account_loan_vendor_payment/models/account_payment.py:8]: You don't need this field, it is 1 or 0, and you can just use loan_id

File: account_loan_vendor_payment/models/account_move.py
Comment (etobella) [inline account_loan_vendor_payment/models/account_move.py:21]: Which is the goal of this function? The loan field is already there.

**Review by etobella — CHANGES_REQUESTED** [review]
I don't get the goal of this module. Are you trying to generate the loan when the payment is received? Usually it is done in a different way, we negotiate with the bank and once we (and they) agree, we receive the payment.

**Comment by francesco-ooops** [comment]
> I don't get the goal of this module. Are you trying to generate the loan when the payment is received? Usually it is done in a different way, we negotiate with the bank and once we (and they) agree, we receive the payment.

Hi @etobella, the flow here is: 

- for a vendor bill, user negotiates with the bank for a loan to pay it
- once the loan is approved, payment is posted
- contextually, loan is opened

With this flow, loan is not created until approved by bank, and once this happens both loan and payment are created.

**Pull request closed without merging**
